test/unit/test_manifest_download.py

```diff
@@ -0,0 +1,82 @@
+import os
+import shutil
+import tempfile
+import unittest
+
+import scandir
+from mock import patch
+
+from hca.dss import DSSClient
+
+
+def _touch_file(path):
+    try:
+        os.makedirs(os.path.split(path)[0])
+    except OSError as e:
+        if 'File exists' in str(e):
+            pass
+    with open(path, 'w'):
+        pass
+
+
+def _fake_download(*args, **kwargs):
+    _touch_file(args[2])
+
+
+class TestManifestDownload(unittest.TestCase):
+    manifest = [
+        ['bundle_uuid', 'bundle_version', 'file_content_type', 'file_name', 'file_sha256', 'file_size', 'file_uuid', 'file_version', 'file_indexed'],
+        ['a_uuid', '1_version', 'somestuff', 'a_file_name', 'ad3fc1e4898e0bce096be5151964a81929dbd2a92bd5ed56a39a8e133053831d', '12', 'af_uuid', 'af_version', 'False'],
+        ['b_uuid', '1_version', 'somestuff', 'b_file_name', '8f35071eaeedd9d6f575a8b0f291daeac4c1dfdfa133b5c561232a00bf18c4b4', '2', 'bf_uuid', 'af_version', 'False'],
+        ['c_uuid', '1_version', 'somestuff', 'c_file_name', '8f3404db04bdede03e9128a4b48599d0ecde5b2e58ed9ce52ce84c3d54a3429c', '41', 'cf_uuid', 'af_version', 'False'],
+    ]
+
+    def setUp(self):
+        self.prev_wd = os.getcwd()
+        self.tmp_dir = tempfile.mkdtemp()
+        os.chdir(self.tmp_dir)
+        self.dss = DSSClient()
+        with open('manifest.tsv', 'w') as f:
+            f.write('\n'.join(['\t'.join(row) for row in self.manifest]))
+
+    def tearDown(self):
+        os.chdir(self.prev_wd)
+        shutil.rmtree(self.tmp_dir)
+
+    def _assert_all_files_downloaded(self):
+        num_files = sum(len(files) for _, _, files in scandir.walk('.'))
+        self.assertEqual(num_files, len(self.manifest))  # -1 for header, but +1 for manifest.tsv in cwd
+        expected_file_paths = [self.dss._file_path(row[4]) for row in self.manifest[1:]]
```

Comment:
We shouldn't rely on code under test to compute an assertion. If there is a bug in the code under test, the assertion will be wrong. I want to see explicit paths like `3f/4c6d/3f4c6d…` here in a literal assertion. I'd rather have a test with tighter coupling to the code under test, and that is therefore more brittle (false negatives if the code under test is changed) than a "smart" assertion that risks giving false positives if the code under test is changed.

Reply:
This is a better approach, but the '/'s aren't portable (right?) I could write a bunch of path.joins to get the file's path, but then I end up basically writing `_file_path()` all over again. It seems like it's not worth it. What do you think?

Reply:
Nvm, I'll just use `os.sep` and a lot of plus signs.